core/app_client.go

```diff
@@ -0,0 +1,163 @@
+	retData, err := a.LaundryAdapter.SubmitServiceRequest(machineID, problemCode, comments, firstname, lastname, phone, email, conf)
+	if err != nil {
+		return nil, err
+	}
+	return retData, nil
+}
+
+func (a appClient) GetBuilding(bldgID string, adaOnly bool, latitude float64, longitude float64) (*model.Building, error) {
+	conf, _ := a.app.GetEnvConfigs()
+	retData, err := a.LocationAdapter.GetBuilding(bldgID, adaOnly, latitude, longitude, conf)
+	if err != nil {
+		return nil, err
+	}
+	return retData, nil
+
+}
+
+func (a appClient) GetEntrance(bldgID string, adaOnly bool, latitude float64, longitude float64) (*model.Entrance, error) {
+	conf, _ := a.app.GetEnvConfigs()
+	retData, err := a.LocationAdapter.GetEntrance(bldgID, adaOnly, latitude, longitude, conf)
+	if err != nil {
+		return nil, err
+	}
+	return retData, nil
+
+}
+
+func (a appClient) GetBuildings() (*[]model.Building, error) {
+	conf, _ := a.app.GetEnvConfigs()
+	retData, err := a.LocationAdapter.GetBuildings(conf)
+	if err != nil {
+		return nil, err
+	}
+	return retData, nil
+
+}
+
+func (a appClient) GetContactInfo(uin string, accessToken string, mode string) (*model.Person, int, error) {
+	conf, _ := a.app.GetEnvConfigs()
+	retData, statuscode, err := a.ContactAdapter.GetContactInformation(uin, accessToken, mode, conf)
+	if err != nil {
+		return nil, statuscode, err
+	}
+	return retData, statuscode, nil
+}
+
+func (a appClient) GetGiesCourses(uin string, accessToken string) (*[]model.GiesCourse, int, error) {
+	conf, _ := a.app.GetEnvConfigs()
+	retData, statuscode, err := a.Courseadapter.GetGiesCourses(uin, accessToken, conf)
+	if err != nil {
+		return nil, statuscode, err
+	}
+	return retData, statuscode, nil
+
+}
+
+func (a appClient) GetStudentCourses(uin string, termid string, accessToken string) (*[]model.Course, int, error) {
+	conf, _ := a.app.GetEnvConfigs()
+	retData, statuscode, err := a.Courseadapter.GetStudentCourses(uin, termid, accessToken, conf)
+	if err != nil {
+		return nil, statuscode, err
+	}
+	return retData, statuscode, nil
+}
+
+func (a appClient) GetTermSessions() (*[4]model.TermSession, error) {
+
+	retData, err := a.Courseadapter.GetTermSessions()
+	if err != nil {
+		return nil, err
+	}
+	return retData, nil
+}
+
+// newAppClient creates new appClient
+func newAppClient(app *Application) appClient {
+
+	client := appClient{app: app}
+	//read assets
+	file, _ := ioutil.ReadFile("./assets/assets.json")
+	assets := model.Asset{}
+	_ = json.Unmarshal([]byte(file), &assets)
+	laundryAssets := make(map[string]model.LaundryDetails)
+
+	for i := 0; i < len(assets.Laundry.Assets); i++ {
+		laundryAsset := assets.Laundry.Assets[i]
+		laundryAssets[laundryAsset.LocationID] = laundryAsset.Details
+	}
+
+	client.ContactAdapter = uiucadapters.NewUIUCContactAdapter()
+	client.LaundryAdapter = uiucadapters.NewCSCLaundryAdapter(laundryAssets)
+	client.Courseadapter = uiucadapters.NewCourseAdapter()
+	client.LocationAdapter = uiucadapters.NewUIUCWayFinding()
```

Comment:
Ideally we should pass these adapters as the point of using the interfaces is to avoid tightly coupling the internal login in the `core` package to a specific driver/driven adapter. In general if you ever find yourself importing something from `driven` or `driver` into the `core` package (or each other) this design principal is being violated. 

To put this in more concrete terms, imagine we wanted to hook up new driven adapters for a different university. We shouldn't have to change any code in the `core` package at all to do this since these components should be isolated within the `driven` packages.

If you have any questions about this design principal, please let me know. For reference, this is the "hexagonal architecture" that we sometimes mention.

Reply:
Will address when bringing in the identity building block